Merge Multi-Arch Manifest is broken on main

https://github.com/OpenHands/software-agent-sdk/actions/runs/19073878155/job/54484406142

```
Run docker/login-action@v3
Logging into ghcr.io...
Login Succeeded!
3s
Run SHORT_SHA=7cdc067
Creating multi-arch manifest: ghcr.io/openhands/agent-server:7cdc067-python
#1 [internal] pushing ghcr.io/openhands/agent-server:7cdc067-python
#1 0.000 pushing sha256:c9fd19e06cd0c0190a95f802b07416f35323dc4abfa4b50c932f1e1b61efaa00 to ghcr.io/openhands/agent-server:7cdc067-python
#1 DONE 0.6s
Inspecting multi-arch manifest:
Name:      ghcr.io/openhands/agent-server:7cdc067-python
MediaType: application/vnd.oci.image.index.v1+json
Digest:    sha256:c9fd19e06cd0c0190a95f802b07416f35323dc4abfa4b50c932f1e1b61efaa00
           
Manifests: 
  Name:        ghcr.io/openhands/agent-server:7cdc067-python@sha256:991ecbbb278118510d0755e9d499b3706c7fb01ae401d87cb80c7649ef47686c
  MediaType:   application/vnd.docker.distribution.manifest.v2+json
  Platform:    linux/amd64
               
  Name:        ghcr.io/openhands/agent-server:7cdc067-python@sha256:f7e2762b423d93af13466f688c696f60dc8689ff6e0ea82b653827738947ddc0
  MediaType:   application/vnd.oci.image.manifest.v1+json
  Platform:    linux/arm64
               
  Name:        ghcr.io/openhands/agent-server:7cdc067-python@sha256:6f6706ce6a53043454ec4f16c866a306bcaf68bf36874794816f8262e3c4520f
  MediaType:   application/vnd.oci.image.manifest.v1+json
  Platform:    unknown/unknown
  Annotations: 
    vnd.docker.reference.digest: sha256:f7e2762b423d93af13466f688c696f60dc8689ff6e0ea82b653827738947ddc0
    vnd.docker.reference.type:   attestation-manifest
✓ Multi-arch manifest created: ghcr.io/openhands/agent-server:7cdc067-python
Creating latest multi-arch manifest: ghcr.io/openhands/agent-server:latest-python
ERROR: ghcr.io/openhands/agent-server:latest-python-arm64: not found
Error: Process completed with exit code 1.
```

--- a/.github/workflows/server.yml
+++ b/.github/workflows/server.yml
@@ -374,8 +374,8 @@ jobs:
                       LATEST_TAG="latest-${VARIANT}"
                       echo "Creating latest multi-arch manifest: ${IMAGE}:${LATEST_TAG}"
                       docker buildx imagetools create -t ${IMAGE}:${LATEST_TAG} \
-                        ${IMAGE}:latest-${VARIANT}-amd64 \
-                        ${IMAGE}:latest-${VARIANT}-arm64
+                        ${IMAGE}:main-${VARIANT}-amd64 \
+                        ${IMAGE}:main-${VARIANT}-arm64
                       
                       echo "Inspecting latest multi-arch manifest:"
                       docker buildx imagetools inspect ${IMAGE}:${LATEST_TAG}

